### ISSUE TITLE
Using iterator in loop prevents destruction of another value

### ISSUE THREAD

**Issue by inv2004** [nim-lang/Nim#14238]
Using iterator in loop prevents to destroy another value

https://github.com/inv2004/kdb_nim/blob/b39f6b4f02ffd1f083f7ab1d1bdde2ee4048581f/tests/mem.nim#L42

### Example
```nim
test "testMemFailed":
  for i in 0..100:
    echo "DEBUG1"
    let vv = K(k: knk(0))
    for i in vv.k:
      discard
    let x = K(k: ktn(11, 0))
    echo "DEBUG2"
```
```
git clone https://github.com/inv2004/kdb_nim
cd kdb_nim
git checkout b39f6b4f02ffd1f083f7ab1d1bdde2ee4048581f

nim c --gc:arc  -d:useMalloc -d:danger tests/test1.nim
./tests/test1
```

### Current Output
```
DEBUG1
destroy K: kList rc = 0 addr =
DEBUG2
DEBUG1
destroy K: kList rc = 0 addr =
DEBUG2
... repeat 100x times
destroy K: kVecSym rc = 0 addr =
destroy K: kList rc = 0 addr =
```

### Expected Output
```
DEBUG1
destroy K: kList rc = 0 addr =
destroy K: kVecSym rc = 0 addr =
DEBUG2
... repeat 100x times
```

### Possible Solution

I do not know. Workaround - to put loop code in function

### Additional Information

Strange thing:
* this "for" prevents next's line value to be destroyed in loop
If you comment "for" - you can find correct output

```
$ nim -v
Nim Compiler Version 1.2.0 [Linux: amd64]          # Windows WSL
```

**Comment by cooldome** [comment]
Without `K` definition this test case hard to investigate. Could you please submit a full self contained test case.

**Comment by inv2004** [comment]
@cooldome there is a link to self-contained repo + git commands and run the specific test. Please let me know if it not enough.

**Comment by Clyybber** [comment]
Fixed by #14263 :D
